src/ts/components/settingspanel.ts

```diff
@@ -300,8 +316,8 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
     this.animateNavigation(targetPage, sourcePage, skipAnimation);
 
     this.updateActivePageClass();
-    targetPage.onActiveEvent();
     sourcePage.onInactiveEvent();
+    targetPage.onActiveEvent();
```

Comment:
The previous order was 'wrong' and caused an issue with the new dynamic `SettingsPanelPage` approach, where the previous page would get the inactive event after the new page got the active event.